diff --git a/packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssao2RenderingPipeline.pure.ts b/packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssao2RenderingPipeline.pure.ts
index b9f52fcfc5..316b1760c8 100644
--- a/packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssao2RenderingPipeline.pure.ts
+++ b/packages/dev/core/src/PostProcesses/RenderPipeline/Pipelines/ssao2RenderingPipeline.pure.ts
@@ -365,14 +365,10 @@ export class SSAO2RenderingPipeline extends PostProcessRenderPipeline {
             if (!this._forcedGeometryBuffer) {
                 scene.enableGeometryBufferRenderer();
             }
-            if (scene.geometryBufferRenderer?.generateNormalsInWorldSpace) {
-                Logger.Error("SSAO2RenderingPipeline does not support generateNormalsInWorldSpace=true for the geometry buffer renderer!");
-            }
+            this._thinSSAORenderingPipeline._ssaoPostProcess.normalsInWorldSpace = !!this._geometryBufferRenderer?.generateNormalsInWorldSpace;
         } else {
             scene.enablePrePassRenderer();
-            if (scene.prePassRenderer?.generateNormalsInWorldSpace) {
-                Logger.Error("SSAO2RenderingPipeline does not support generateNormalsInWorldSpace=true for the prepass renderer!");
-            }
+            this._thinSSAORenderingPipeline._ssaoPostProcess.normalsInWorldSpace = !!scene.prePassRenderer?.generateNormalsInWorldSpace;
         }
 
         this._originalColorPostProcess = new PassPostProcess("SSAOOriginalSceneColor", 1.0, null, Texture.BILINEAR_SAMPLINGMODE, scene.getEngine(), undefined, this._textureType);
diff --git a/packages/dev/core/src/PostProcesses/thinSSAO2PostProcess.ts b/packages/dev/core/src/PostProcesses/thinSSAO2PostProcess.ts
index e41f014aba..3ee178db48 100644
--- a/packages/dev/core/src/PostProcesses/thinSSAO2PostProcess.ts
+++ b/packages/dev/core/src/PostProcesses/thinSSAO2PostProcess.ts
@@ -35,6 +35,7 @@ export class ThinSSAO2PostProcess extends EffectWrapper {
         "maxZ",
         "minZAspect",
         "depthProjection",
+        "normalWorldToView",
     ];
 
     public static readonly Samplers = ["randomSampler", "depthSampler", "normalSampler"];
@@ -99,6 +100,20 @@ export class ThinSSAO2PostProcess extends EffectWrapper {
 
     private _epsilon: number = 0.02;
 
+    private _normalsInWorldSpace = false;
+
+    public set normalsInWorldSpace(value: boolean) {
+        if (this._normalsInWorldSpace === value) {
+            return;
+        }
+        this._normalsInWorldSpace = value;
+        this.updateEffect();
+    }
+
+    public get normalsInWorldSpace(): boolean {
+        return this._normalsInWorldSpace;
+    }
+
     public set epsilon(n: number) {
         this._epsilon = n;
         this.updateEffect();
@@ -180,6 +195,9 @@ export class ThinSSAO2PostProcess extends EffectWrapper {
             effect.setFloat4("viewport", orthoLeft, orthoRight, orthoBottom, orthoTop);
         }
         effect.setMatrix("projection", projectionMatrix);
+        if (this._normalsInWorldSpace) {
+            effect.setMatrix("normalWorldToView", camera.getViewMatrix());
+        }
 
         effect.setTexture("randomSampler", this._randomTexture);
     }
@@ -265,6 +283,9 @@ export class ThinSSAO2PostProcess extends EffectWrapper {
         if (this.camera?.mode === Camera.ORTHOGRAPHIC_CAMERA) {
             defines += `\n#define ORTHOGRAPHIC_CAMERA`;
         }
+        if (this._normalsInWorldSpace) {
+            defines += `\n#define NORMAL_WORLDSPACE`;
+        }
 
         return defines;
     }
diff --git a/packages/dev/core/src/Shaders/ssao2.fragment.fx b/packages/dev/core/src/Shaders/ssao2.fragment.fx
index 4486112c4e..0c805642cb 100644
--- a/packages/dev/core/src/Shaders/ssao2.fragment.fx
+++ b/packages/dev/core/src/Shaders/ssao2.fragment.fx
@@ -45,18 +45,24 @@ varying vec2 vUV;
 	uniform mat3 depthProjection;
 	uniform float maxZ;
 	uniform float minZAspect;
-	uniform vec2 texelSize;
-
-	uniform mat4 projection;
-
-	void main()
-	{
+	uniform vec2 texelSize;
+
+	uniform mat4 projection;
+#ifdef NORMAL_WORLDSPACE
+	uniform mat4 normalWorldToView;
+#endif
+
+	void main()
+	{
 		vec3 random = textureLod(randomSampler, vUV * randTextureTiles, 0.0).rgb;
 		float depth = textureLod(depthSampler, vUV, 0.0).r;
-		float depthSign = sign(depth);
-		depth = depth * depthSign;
-		vec3 normal = textureLod(normalSampler, vUV, 0.0).rgb;
-		float occlusion = 0.0;
+		float depthSign = sign(depth);
+		depth = depth * depthSign;
+		vec3 normal = textureLod(normalSampler, vUV, 0.0).rgb;
+#ifdef NORMAL_WORLDSPACE
+		normal = normalize((normalWorldToView * vec4(normal, 0.0)).xyz);
+#endif
+		float occlusion = 0.0;
 		float correctedRadius = min(radius, minZAspect * depth / near);
 
 	#ifdef ORTHOGRAPHIC_CAMERA
diff --git a/packages/dev/core/src/ShadersWGSL/ssao2.fragment.fx b/packages/dev/core/src/ShadersWGSL/ssao2.fragment.fx
index 8f0be0ad8d..0621537e48 100644
--- a/packages/dev/core/src/ShadersWGSL/ssao2.fragment.fx
+++ b/packages/dev/core/src/ShadersWGSL/ssao2.fragment.fx
@@ -48,18 +48,24 @@ var textureSampler: texture_2d<f32>;
 	uniform depthProjection: mat3x3f;
 	uniform maxZ: f32;
 	uniform minZAspect: f32;
-	uniform texelSize: vec2f;
-
-	uniform projection: mat4x4f;
-
-	@fragment
-	fn main(input: FragmentInputs) -> FragmentOutputs {
+	uniform texelSize: vec2f;
+
+	uniform projection: mat4x4f;
+#ifdef NORMAL_WORLDSPACE
+	uniform normalWorldToView: mat4x4f;
+#endif
+
+	@fragment
+	fn main(input: FragmentInputs) -> FragmentOutputs {
 		var random: vec3f = textureSampleLevel(randomSampler, randomSamplerSampler, input.vUV * uniforms.randTextureTiles, 0.0).rgb;
 		var depth: f32 = textureSampleLevel(depthSampler, depthSamplerSampler, input.vUV, 0.0).r;
-		var depthSign: f32 = sign(depth);
-		depth = depth * depthSign;
-		var normal: vec3f = textureSampleLevel(normalSampler, normalSamplerSampler, input.vUV, 0.0).rgb;
-		var occlusion: f32 = 0.0;
+		var depthSign: f32 = sign(depth);
+		depth = depth * depthSign;
+		var normal: vec3f = textureSampleLevel(normalSampler, normalSamplerSampler, input.vUV, 0.0).rgb;
+#ifdef NORMAL_WORLDSPACE
+		normal = normalize((uniforms.normalWorldToView * vec4f(normal, 0.0)).xyz);
+#endif
+		var occlusion: f32 = 0.0;
 		var correctedRadius: f32 = min(uniforms.radius, uniforms.minZAspect * depth / uniforms.near);
 
 	#ifdef ORTHOGRAPHIC_CAMERA
